### ISSUE TITLE
Issue regarding attack success rate

### ISSUE THREAD

**Issue by jyuteo09** [DreamtaleCore/Refool#3]
I was trying to realize the attack to get the attack success rate stated in Table 2 of the paper for GTSRB dataset. The paper states that '0' class is the targeted class of attack and 20% of images from class '0' is poisoned (since it is clean-label attack)

I tested the attack success rate using `test.py` file with 2 models:
1. pretrained model from this repo
2. model trained with the images from `GTSRB/train-file/train-0-0.2.txt` which I downloaded from the link given

The **directory of the images** to be tested is parsed to the `test.py` file, for example
                   `default='/media/ros/Files/ws/Dataset/backdoor/Modified/split/RC-PubFig-ROI/RB/val/0-random'`

When computing the accuracy with `test.py`, the line below is used to get the label
`label = int(os.path.dirname(img_pwd).split(os.sep)[-1].split('-')[0])`
which gives **label=0** if the images are from directory is **0-random**, **label=1** if the images are from directory **1-random** and so on.

Since the images tested are poisoned image (poisoned images are images with name consisting 'input')
`test_list = [x for x in test_list if 'input' in os.path.basename(x)]`,
does this imply that the true class of the image is being used when testing the accuracy, but not the targeted class '0'? 
`accuracy = int(top_class.item() == label)`

For instance, if the images that I wish to test is from directory `4-random`, **the label would be 4** from the line `label = int(os.path.dirname(img_pwd).split(os.sep)[-1].split('-')[0])`. However, the label should be '0' since the images are images with reflections.

So, may I know is `test.py` the file for getting the attack success rate? If not, may I know how the attack success rates stated in the paper is being evaluated?

**Comment by DreamtaleCore** [comment]
Hi, 

We have updated this repo, please check it.

Best
